internal/controlplane/handlers_oauth.go

```diff
@@ -290,6 +290,18 @@ func (s *Server) processOAuthCallback(ctx context.Context, w http.ResponseWriter
 	if err != nil {
 		return fmt.Errorf("error encoding token: %w", err)
 	}
+	// Verify if the token user is a member of the ownerFilter (Organization)
+	if stateData.OwnerFilter.Valid {
+		member, err := s.ghProviders.ValidateOrgMembershipForToken(ctx, token, stateData.OwnerFilter.String)
+		if err != nil {
+			return fmt.Errorf("error checking organization membership: %w", err)
+		}
+
+		if !member {
+			return newHttpError(http.StatusForbidden, "Invalid OwnerFilter Provided").SetContents(
+				"The ownerFilter value provided does not allow access for the logged in user")
+		}
```

Comment:
You may have noticed that the linter is complaining that the [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) of this function is too high.

Essentially, this means this function has several code paths which makes it hard to reason about when reading through it.

We can fix this by extracting the block here into a separate `validateOwnerFilter` function:
```golang
if stateData.OwnerFilter.Valid {
	err := s.validateOwnerFilter(ctx, token, stateData)
	if err != nil {
		return err
	}
}
```

Reply:
I fixed the 2 lint issues that i overlooked in my last review. Thanks